src/MongoDB.Analyzer/Core/Utilities/SyntaxNodeExtensions.cs

```diff
@@ -16,6 +16,46 @@ namespace MongoDB.Analyzer.Core;
 
 internal static class SyntaxNodeExtensions
 {
+    public static MemberDeclarationSyntax AddBsonAttributes(this MemberDeclarationSyntax memberDeclaration, IEnumerable<AttributeData> attributes)
```

Comment:
Check if `T AddBsonAttributes<T>(this T memberDeclaration, ...) where T : MemberDeclarationSyntax` works.
might eliminate the need for casting as FieldDeclarationSyntax, and ensure type safety.

Reply:
Done